library/src/main/java/me/grantland/widget/AutofitHelper.java

```diff
@@ -128,19 +131,49 @@ private static void autofit(TextView view, TextPaint paint, float minTextSize, f
                     displayMetrics);
         }
 
-        if (size < minTextSize) {
-            size = minTextSize;
+        if (mIsHeightFitting) {
+            int targetHeight = view.getHeight() - view.getPaddingTop() - view.getPaddingBottom();
+            if (targetHeight <= 0) {
+                if (size < minTextSize) {
+                    size = minTextSize;
+                }
+                view.setTextSize(TypedValue.COMPLEX_UNIT_PX, size);
+                return;
+            }
+
+            float textHeight = getTextHeight(text, paint, targetWidth, size);
+            float heightRatio = targetHeight / textHeight;
+            float newSize = size * heightRatio;
+            if (newSize < size) {
+                size = newSize;
+            }
```

Comment:
I'm actually not super familiar with StaticLayout; that's just what KonradJanica used to calculate height. However, I think this article on [The Mathematics of Golden Ratio Typography](http://www.pearsonified.com/2011/12/golden-ratio-typography.php) sheds some light on the mathematics between text height, text width, and text size.